File: ts/wam/7096.ts
export const WamDefenseModeClick = 7096 as const

export interface DefenseModeClickEvent {
  /** field 1, wire `control_name` */
  controlName?: DEFENSEMODECLICKCONTROLNAME
  /** field 2, wire `defense_mode_click_accepted` */
  defenseModeClickAccepted?: boolean
  /** field 3, wire `desired_state` */
  desiredState?: DEFENSEMODECLICKDESIREDSTATE
}

File: rust/wam/7096.rs
pub const WAM_DEFENSE_MODE_CLICK: u32 = 7096;

#[derive(Debug, Default)]
pub struct DefenseModeClickEvent {
    /// field 1, wire `control_name`
    pub control_name: Option<DEFENSEMODECLICKCONTROLNAME>,
    /// field 2, wire `defense_mode_click_accepted`
    pub defense_mode_click_accepted: Option<bool>,
    /// field 3, wire `desired_state`
    pub desired_state: Option<DEFENSEMODECLICKDESIREDSTATE>,
}